wordpress-mobile/WordPress-iOS
# Caption shortcode support

## Description

Adds caption support to WPiOS through Aztec.

Fixes #7715 

## Details

To implement Caption support into WPiOS, this PR migrates some of the code we already have in Aztec's Example app.

What changed:

* New UI support for editing image captions in the image details view.
* Added `CaptionShortcodePreProcessor.swift` and `CaptionShortcodePostProcessor.swift` to handle the shortcode conversions to and from HTML.

## Testing

* Make sure you test editing both the caption and ALT fields.  Clear the fields.  Make sure any change is saved.
* Edit caption styles by right clicking on the text and using iOS's built-in styling options.
* Make sure captions are rendered.
* Make sure the HTML view shows captions properly.
* Make sure `<figure>` html elements are properly parsed too.  Pase the following snippet in HTML mode and switch back to visual:

```html
<figure>
  <img src="img_pulpit.jpg" alt="The Pulpit Rock" width="304" height="228">
  <figcaption>Fig1. - A view of the pulpit rock in Norway.</figcaption>
</figure>
```
